mico-admin/src/app/api/models.service.ts

```diff
@@ -2,13 +2,22 @@ import { Injectable } from '@angular/core';
 import { AsyncSubject, Observable, of, from } from 'rxjs';
 import { ApiModel, ApiModelAllOf, ApiModelRef } from './apimodel';
 import { concatMap, reduce, first, timeout, map } from 'rxjs/operators';
-import { freezeObject } from './api.service';
+import { ApiService, freezeObject } from './api.service';
 
 @Injectable({
     providedIn: 'root'
 })
 export class ModelsService {
 
+    private remoteModels;
+
+    constructor(private apiService: ApiService, ) {
+        // TODO consider unsubscribing
+        apiService.getModelDefinitions().subscribe(val => {
```

Comment:
Consider using the Observable directly instead of storing the provided value to avoid cases where remoteModels is still null but a model is already requested by the ui.